File: nhglib-core/src/io/github/movementspeed/nhglib/input/models/InputSource.java
package io.github.movementspeed.nhglib.input.models;

/**
 * Created by Fausto Napoli on 22/01/2017.
 */
public class InputSource {
    private String name;
    private Object value;

    public void setName(String name) {
        this.name = name;
    }

    public void setValue(Object value) {
        this.value = value;
    }

    public Boolean is(String name) {
        return name.contentEquals(name);
    }

    public String getName() {
        return name;
    }

    public Object getValue() {
        return value;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/input/models/InputSource.kt
package io.github.movementspeed.nhglib.input.models

/**
 * Created by Fausto Napoli on 22/01/2017.
 */
class InputSource {
    var name: String? = null
    var value: Any? = null

    fun `is`(name: String): Boolean {
        return name.contentEquals(name)
    }
}
